docs/src/main/paradox/scala/http/server-side/http2.md

```diff
@@ -0,0 +1,83 @@
+# HTTP/2 (Preview)
+
+HTTP/2 support in akka-http is currently available as a preview.
+This means it is ready to be evaluated, but the API's and behavior is likely to change.
+
+@@@ div { .group-scala }
+
+## Enabling HTTP/2
+
+sbt
+:   @@@vars
+    ```sbt
+    "com.typesafe.akka" %% "akka-http2-support" % "$project.version$"
+    ```
+    @@@
+
+maven
+:   @@@vars
+    ```
+    <dependency>
+        <groupId>com.typesafe.akka</groupId>
+        <artifactId>akka-http2-support_2.12</artifactId>
+        <version>$project.version$</version>
+    </dependency>
+    ```
+    @@@
+
+We only support HTTP/2 over HTTPS: browsers don't typically support HTTP/2 on plain HTTP, and other clients that do are rare.
+
+To explicitly bind a server that accepts both HTTP and HTTP/2, simply use @scaladoc[Http2().bindAndHandleAsync](akka.http.scaladsl.Http2Ext) in the same way you would @scaladoc[Http().bindAndHandleAsync](akka.http.scaladsl.HttpExt):
+
+Scala
+:   @@snip[Http2Spec.scala](../../../../../test/scala/docs/http/scaladsl/Http2Spec.scala) { #bindAndHandleAsync }
+
+Java
+:   @@snip[Http2Test.java](../../../../../test/java/docs/http/javadsl/Http2Test.java) { #bindAndHandleAsync }
+
+For an even more transparent experience, you can use the regular @scaladoc[Http().bindAndHandleAsync](akka.http.scaladsl.HttpExt) and enable HTTP/2 through configuration, but make sure you include the dependency and use the `Async` variation of `bindAndHandle`:
+
+```
+akka.http.server.preview.enable-http2 = on
+```
+
+At this point you should be able to connect, which you can verify with `curl -v`:
```

Comment:
And link to https://simonecarletti.com/blog/2016/01/http2-curl-macosx/ which Konrad found